Source/santasyncservice/SNTPushClientNATSCommands.mm

```diff
@@ -0,0 +1,189 @@
+                                      SNTPushClientNATS *self) {
+  if (!self || !msg) {
+    if (msg) {
+      natsMsg_Destroy(msg);
+    }
+    return;
+  }
+
+  if (self.isShuttingDown) {
+    natsMsg_Destroy(msg);
+    return;
+  }
+
+  NSString *msgSubject = @(natsMsg_GetSubject(msg) ?: "<unknown>");
+  NSString *replyTopic = natsMsg_GetReply(msg) ? @(natsMsg_GetReply(msg)) : nil;
+
+  LOGD(@"NATS: Received command message on subject '%@' with reply '%@'", msgSubject,
+       replyTopic ?: @"<no reply>");
+
+  if (!replyTopic) {
+    LOGW(@"NATS: Command message on %@ has no reply topic, ignoring", msgSubject);
+    natsMsg_Destroy(msg);
+    return;
+  }
+
+  // Validate message data before dispatching
+  const void *data = natsMsg_GetData(msg);
+  int dataLen = natsMsg_GetDataLength(msg);
+
+  if (!data || dataLen <= 0) {
+    LOGE(@"NATS: Command message on %@ has no data", msgSubject);
+    // Try to send error response, but don't fail if that also fails
+    santa::commands::v1::SantaCommandResponse errorResponse;
+    errorResponse.set_code(santa::commands::v1::SANTA_COMMAND_RESPONSE_CODE_ERROR);
+    [self publishResponse:errorResponse toReplyTopic:replyTopic];
+    natsMsg_Destroy(msg);
+    return;
+  }
+
+  // Deserialize the message to SantaCommandRequest
+  // Note: We must extract all data from msg before destroying it, as NATS owns the message
+  // and will free it after this callback returns
+  santa::commands::v1::SantaCommandRequest command;
+  if (!command.ParseFromArray(data, dataLen)) {
+    LOGE(@"NATS: Failed to parse SantaCommandRequest from message on %@", msgSubject);
+    // Try to send error response, but don't fail if that also fails
+    santa::commands::v1::SantaCommandResponse errorResponse;
+    errorResponse.set_code(santa::commands::v1::SANTA_COMMAND_RESPONSE_CODE_ERROR);
+    [self publishResponse:errorResponse toReplyTopic:replyTopic];
+    natsMsg_Destroy(msg);
+    return;
+  }
+
+  // Destroy the message now - NATS owns it and will free it after callback returns anyway
+  // We've extracted all needed data (command, replyTopic, msgSubject) which are safe to capture
+  natsMsg_Destroy(msg);
+  msg = NULL;  // Prevent accidental use
+
+  // Process on message queue to serialize handling of messages
+  // Failures are logged but don't crash the client
+  dispatch_async(self.messageQueue, ^{
+    if (self.isShuttingDown) {
+      return;
+    }
+
+    santa::commands::v1::SantaCommandResponse response =
+        [self dispatchSantaCommandToHandler:command];
+
+    // Publish the response
+    [self publishResponse:response toReplyTopic:replyTopic];
+  });
+}
+
+__BEGIN_DECLS
+
+// NATS-compatible wrapper that converts void *closure to SNTPushClientNATS *
+void commandMessageHandler(natsConnection *nc, natsSubscription *sub, natsMsg *msg, void *closure) {
+  SNTPushClientNATS *self = (__bridge SNTPushClientNATS *)closure;
+  commandMessageHandlerImpl(nc, sub, msg, self);
+}
+
+__END_DECLS
```

Comment:
Assuming this doesn't need external linkage...
```suggestion
// NATS-compatible wrapper that converts void *closure to SNTPushClientNATS *
void CommandMessageHandler(natsConnection *nc, natsSubscription *sub, natsMsg *msg, void *closure) {
  SNTPushClientNATS *self = (__bridge SNTPushClientNATS *)closure;
  CommandMessageHandlerImpl(nc, sub, msg, self);
}
```

Reply:
We do need external linkage.